pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -603,15 +603,16 @@ func selectVictimsOnNode(
 		if err := nodeInfo.RemovePod(rp); err != nil {
 			return err
 		}
-		status := ph.RunPreFilterExtensionRemovePod(ctx, state, pod, rp, nodeInfo)
+		status := ph.RunPreFilterExtensionRemovePod(ctx, state, pod, framework.NewPodInfo(rp), nodeInfo)
```

Comment:
It looks like you can change `removePod` to accept a PodInfo

Reply:
also change the following to use podInfo:

1) `potentialVictims` should be of type PodInfo
2) change `filterPodsWithPDBViolation`  to accept and return lists of type PodInfo
3) change `addPod` and `reprievePod` to accept PodInfo

Reply:
It is fine you were planning to do that in a separate PR, but if so, please update the title and description of the PR to indicate that the goal of this PR is to update the `PreFilterExtensions` interface to use PodInfo (not reduce struct initialization).

Reply:
Yes, I am working on it locally with one failing test to investigate. Let us keep this PR small and focused.